### PR TITLE
feat(retention) Enable retention and set max versions for executionRequests

### PR DESCRIPTION
**Pull request by chriscollins3456**
Set the default env variable of ENTITY_SERVICE_ENABLE_RETENTION = true. Also set the default retention config file to allow max 20 for `datahubExecutionRequestResult` aspects. This is because the new live logs feature really expands then number of request result aspects has really expanded.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/war/src/main/resources/boot/retention.yaml [hunk 1]
```diff
@@ -1,9 +1,15 @@
-- entity: "*"
-  aspect: "*"
+- entity: "dataHubExecutionRequest"
+  aspect: "dataHubExecutionRequestResult"
   config:
     retention:
       version:
         maxVersions: 20
```
Comment (shirshanka) [inline metadata-service/war/src/main/resources/boot/retention.yaml:6]: even 20 is a lot just for execution results, we should be fine with a lower number like 5?

**Review by shirshanka — APPROVED** [review]
LGTM

File: metadata-service/war/src/main/resources/boot/retention.yaml (hunk 1 above)
Comment (chriscollins3456) [reply metadata-service/war/src/main/resources/boot/retention.yaml:6]: nice that sounds reasonable to me

Comment (jjoyce0510) [reply metadata-service/war/src/main/resources/boot/retention.yaml:6]: +1

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**